Fatras/include/ActsFatras/Utilities/GammaDistribution.hpp

```diff
@@ -0,0 +1,120 @@
+#include <random>
+
+namespace ActsFatras {
+
+/// Draw random numbers from a gamma distribution.
+/// Does not generate FPE underflow
+///
+/// Implements the same interface as the standard library distributions.
+class GammaDistribution {
+ public:
+  /// Parameter struct that contains all distribution parameters.
+  struct param_type {
+    /// Parameters must link back to the host distribution.
+    using distribution_type = GammaDistribution;
+
+    /// Location parameter.
+    ///
+    /// Shape parameter
+    double alpha = 0.5;
+    /// Scale parameter.
+    double theta = 1.0;
+
+    /// Construct from parameters.
+    param_type(double alpha_, double theta_) : alpha(alpha_), theta(theta_) {}
+    // Explicitly defaulted construction and assignment
+    param_type() = default;
+    param_type(const param_type &) = default;
+    param_type(param_type &&) = default;
+    param_type &operator=(const param_type &) = default;
+    param_type &operator=(param_type &&) = default;
+
+    /// Parameters should be EqualityComparable
+    friend bool operator==(const param_type &lhs, const param_type &rhs) {
+      return (lhs.alpha == rhs.alpha) && (lhs.theta == rhs.theta);
+    }
+  };
+  /// The type of the generated values.
+  using result_type = double;
+
+  /// Construct directly from the distribution parameters.
+  GammaDistribution(double alpha, double theta) : m_cfg(alpha, theta) {}
+  /// Construct from a parameter object.
+  explicit GammaDistribution(const param_type &cfg) : m_cfg(cfg) {}
+  // Explicitlely defaulted construction and assignment
+  GammaDistribution() = default;
+  GammaDistribution(const GammaDistribution &) = default;
+  GammaDistribution(GammaDistribution &&) = default;
+  GammaDistribution &operator=(const GammaDistribution &) = default;
+  GammaDistribution &operator=(GammaDistribution &&) = default;
+
+  /// Reset any possible internal state. Noop, since there is no internal state.
+  void reset() {}
+  /// Return the currently configured distribution parameters.
+  param_type param() const { return m_cfg; }
+  /// Set the distribution parameters.
+  void param(const param_type &cfg) { m_cfg = cfg; }
+
+  /// The minimum value the distribution generates.
+  result_type min() const { return 0.; }
+  /// The maximum value the distribution generates.
+  result_type max() const { return std::numeric_limits<double>::infinity(); }
+
+  /// Generate a random number from the configured gamma distribution.
+  template <typename Generator>
+  result_type operator()(Generator &generator) {
+    return (*this)(generator, m_cfg);
+  }
+  /// Generate a random number from the given gamma distribution.
+  template <typename Generator>
+  result_type operator()(Generator &generator, const param_type &params) {
+    if (params.alpha >= 1.) {
+      std::gamma_distribution<double> gDist(params.alpha, params.theta);
+      return gDist(generator);
+    } else if (params.alpha <= 0.) {
+      return 0.;
+    } else {
```

Comment:
```suggestion
    }
```
matter of taste - you could remove `else` and the `{}` block below which gets rid of some indentation